test/platforms.jl

```diff
@@ -18,6 +18,117 @@ using BinaryBuilderBase: abi_agnostic
                    [Linux(:i686, libc=:glibc), Linux(:i686, libc=:musl), MacOS(:x86_64), Windows(:x86_64)]
 end
 
+@testset "ExtendedPlatform" begin
+    p = ExtendedPlatform(Linux(:x86_64; libc=:musl); microarchitecture = :avx, cuda = "9.2")
+    @test p.p == Linux(:x86_64, libc=:musl)
+    @test p.ext == Dict("microarchitecture" => "avx", "cuda" => "9.2")
```

Comment:
Just for the record, I think I'm going to want the key to be `march` instead of `microarchitecture` just because that's such a mouthful.  Doesn't matter here though.  :P